### PR TITLE
Feature/Zero Decimal Currencies

### LINKED ISSUE #326: Fix currency formatting for currencies without the fractional part
Currencies such as the Japanese Yen JPY and the Korean Won WON don't have the fractional part which seems to break the Nosto currency formatting.

### PR DESCRIPTION
**Pull request by supercid**
Closes #326

### CODE REVIEW

File: Model/Meta/Account/Settings/Currencies/Builder.php [hunk 1]
```diff
@@ -50,6 +50,26 @@ class Builder
     private $eventManager;
     private $currencyFactory;
     private $localeResolver;
+    /* List of zero decimal currencies in compliance with ISO-4217 */
+    private static $zeroDecimalCurrencyCodes = array(
```
Comment (mridang) [inline Model/Meta/Account/Settings/Currencies/Builder.php]: Constant.

File: Model/Meta/Account/Settings/Currencies/Builder.php [hunk 2]
```diff
@@ -143,4 +130,131 @@ public function build(Store $store)
...
+    /**
+     * Returns true if currency is defined to have no decimal part
+     * according to ISO-4217
+     *
+     * @param $currencyCode
+     * @return bool
+     */
+    private function isZeroDecimalCurrency($currencyCode)
+    {
+        foreach (self::$zeroDecimalCurrencyCodes as $zeroDecimalCurrency) {
```
Comment (mridang) [inline Model/Meta/Account/Settings/Currencies/Builder.php]: Why not `in_array `?

File: Model/Meta/Account/Settings/Currencies/Builder.php (hunk 1 above)
Comment (supercid) [reply Model/Meta/Account/Settings/Currencies/Builder.php]: Fixed, thanks

File: Model/Meta/Account/Settings/Currencies/Builder.php (hunk 2 above)
Comment (supercid) [reply Model/Meta/Account/Settings/Currencies/Builder.php]: Slowness, but dismissible in this scenario. Fixed, thanks.

**Pull request merged**